server/controllers/locationController.ts

```diff
@@ -0,0 +1,95 @@
+import { Request, Response } from 'express';
+import { Like } from 'typeorm';
+
+import { Location } from 'server/models/Location';
+
+export default {
+  async index(req: Request, res: Response) {
+    const { country_code, city, region, postal_code, address } = req.query;
```

Comment:
We should validate these fields; e.g. if `city` is undefined it'll look for `%undefined%`.